--- a/cod-tools/src/components/codcif/common.h
+++ b/cod-tools/src/components/codcif/common.h
@@ -10,6 +10,14 @@
 
 #include <stdlib.h>
 
+/* defining ssize_t for MacOS */
+#ifdef __APPLE__
+#ifndef _SSIZE_T
+#define _SSIZE_T
+typedef __darwin_ssize_t ssize_t;
+#endif
+#endif
+
 char *strclone( const char *s );
 char *strnclone( const char *s, size_t length );
 char *strappend( char *s, const char *suffix );
--- a/cod-tools/src/components/codcif/cif2_lexer.h
+++ b/cod-tools/src/components/codcif/cif2_lexer.h
@@ -13,6 +13,14 @@
 #include <cexceptions.h>
 #include <cif_compiler.h>
 
+/* defining ssize_t for MacOS */
+#ifdef __APPLE__
+#ifndef _SSIZE_T
+#define _SSIZE_T
+typedef __darwin_ssize_t ssize_t;
+#endif
+#endif
+
 int cif2lex( void );
 void cif2restart( void );
 
--- a/cod-tools/src/components/codcif/cif_grammar_flex.h
+++ b/cod-tools/src/components/codcif/cif_grammar_flex.h
@@ -12,6 +12,14 @@
 #include <unistd.h> /* for ssize_t */
 #include <cexceptions.h>
 
+/* defining ssize_t for MacOS */
+#ifdef __APPLE__
+#ifndef _SSIZE_T
+#define _SSIZE_T
+typedef __darwin_ssize_t ssize_t;
+#endif
+#endif
+
 typedef enum {
   CIF_FLEX_LEXER_FIX_CTRL_Z = 0x01,
   CIF_FLEX_LEXER_FIX_NON_ASCII_SYMBOLS = 0x02,
