--- a/cod-tools/src/components/codcif/common.h
+++ b/cod-tools/src/components/codcif/common.h
@@ -10,6 +10,14 @@
 
 #include <stdlib.h>
 
+/* defining ssize_t for MacOS */
+#ifdef __APPLE__
+#ifndef _SSIZE_T
+#define _SSIZE_T
+typedef __darwin_ssize_t ssize_t;
+#endif
+#endif
+
 char *strclone( const char *s );
 char *strnclone( const char *s, size_t length );
 char *strappend( char *s, const char *suffix );
