--- a/cod-tools/src/components/codcif/cif2_grammar.y
+++ b/cod-tools/src/components/codcif/cif2_grammar.y
@@ -563,7 +563,7 @@
     int yyretval = 0;
 
     cexception_guard( inner ) {
-        yyin = in;
+        yy2in = in;
         px = &inner; /* catch all parser-generated exceptions */
         if( (yyretval = cif2parse()) != 0 ) {
             if( cif_compiler_cif( cif_cc ) ) {
@@ -579,8 +579,8 @@
         }
     }
     cexception_catch {
-        if( yyin ) {
-            yyin = NULL;
+        if( yy2in ) {
+            yy2in = NULL;
         }
         cexception_reraise( inner, ex );
     }
--- a/cod-tools/src/components/codcif/cif2_lexer.c
+++ b/cod-tools/src/components/codcif/cif2_lexer.c
@@ -21,7 +21,7 @@
 #include <stringx.h>
 #include <assert.h>
 
-FILE *yyin;
+FILE *yy2in;
 
 static CIF_COMPILER *cif_cc;
 
@@ -99,9 +99,9 @@
 
 int cif2lex( void )
 {
-    if( !yyin )
-        yyin = stdin;
-    return cif_lexer( yyin, NULL );
+    if( !yy2in )
+        yy2in = stdin;
+    return cif_lexer( yy2in, NULL );
 }
 
 void cif2restart( void )
--- a/cod-tools/src/components/codcif/yy.h
+++ b/cod-tools/src/components/codcif/yy.h
@@ -9,6 +9,7 @@
 #include <cexceptions.h>
 
 extern FILE *yyin;
+extern FILE *yy2in;
 
 extern int yyparse( void );
 
